src/controllers/import.js

```diff
@@ -126,6 +132,38 @@ function ImportController(context) {
     };
   }
 
+  /**
+   * Get all import jobs between startDate and endDate
+   * @param {object} requestContext - Context of the request.
+   * @param {string} requestContext.params.startDate - The start date of the range.
+   * @param {string} requestContext.params.endDate - The end date of the range.
+   * @param {string} requestContext.pathInfo.headers.x-import-api-key - API key to use for the job.
+   * @returns {Promise<Response>} 200 OK with a JSON representation of the import jobs.
+   */
+  async function getImportJobsByDateRange(requestContext) {
+    const { data: { startDate, endDate }, pathInfo: { headers } } = requestContext;
+    const { 'x-import-api-key': importApiKey } = headers;
+
+    try {
+      // TODO: Once the controller auth changes are implemented, verify the relevant scopes
+      validateImportApiKey(importApiKey);
+      validateIsoDates(startDate, endDate);
+      const jobs = await importSupervisor.getImportJobsByDateRange(startDate, endDate);
+      const importJobs = jobs.map(async (job) => {
+        const hashedApiKey = hashWithSHA256(importApiKey);
+        const { name, imsOrgId, imsUserId } = await dataAccess
```

Comment:
Thinking about reducing our costs by denormalizing this data: @solaris007 what is your preference here? Should the data we need by included in the import-jobs table to avoid this additional dataAccess request for each record?

Reply:
@swetabar, @solaris007 and I spoke about this today. To help performance and decrease the number of queries needed for this endpoint, we would like to see the additional metadata needed for this endpoint included (denormalized) on the import-jobs entities. 

We think this will work well because they api-key entities are idempotent. Do you have any concerns with this approach?

Reply:
I agree as it will reduce costs for apiKey queries significantly.